src/google-maps/google-map-marker/google-map-marker.ts

```diff
@@ -0,0 +1,384 @@
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.mouseover
+   */
+  @Output() mapMouseover = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.mouseup
+   */
+  @Output() mapMouseup = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.position_changed
+   */
+  @Output() positionChanged = new EventEmitter<void>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.rightclick
+   */
+  @Output() mapRightclick = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.shape_changed
+   */
+  @Output() shapeChanged = new EventEmitter<void>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.title_changed
+   */
+  @Output() titleChanged = new EventEmitter<void>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.visible_changed
+   */
+  @Output() visibleChanged = new EventEmitter<void>();
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/marker#Marker.zindex_changed
+   */
+  @Output() zindexChanged = new EventEmitter<void>();
+
+  private readonly _options =
+      new BehaviorSubject<google.maps.MarkerOptions>(DEFAULT_MARKER_OPTIONS);
+  private readonly _title = new BehaviorSubject<string|undefined>(undefined);
+  private readonly _position = new BehaviorSubject<google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _label =
+      new BehaviorSubject<string|google.maps.MarkerLabel|undefined>(undefined);
+  private readonly _clickable = new BehaviorSubject<boolean|undefined>(undefined);
+
+  private readonly _map = new ReplaySubject<google.maps.Map>(1);
+
+  private readonly _destroy = new Subject<void>();
+
+  private readonly _listeners: google.maps.MapsEventListener[] = [];
+
+  private _marker?: google.maps.Marker;
+  private _hasMap = false;
+
+  ngOnInit() {
+    const combinedOptionsChanges = this._combineOptions();
+
+    combineLatest(this._map, combinedOptionsChanges)
+        .pipe(takeUntil(this._destroy))
+        .subscribe(([googleMap, options]) => {
+          if (this._marker) {
+            this._marker.setOptions(options);
+          } else {
+            this._marker = new google.maps.Marker(options);
+            this._marker.setMap(googleMap);
+            this._initializeEventHandlers();
+          }
+        });
+  }
+
+  ngOnDestroy() {
+    this._destroy.next();
+    this._destroy.complete();
+    for (let listener of this._listeners) {
+      listener.remove();
+    }
+    if (this._marker) {
+      this._marker.setMap(null);
+    }
+  }
+
+  setMap(googleMap: google.maps.Map) {
```

Comment:
Should this method be marked as internal (underscore prefix)?

Reply:
It's called from the google-maps component, so it's not strictly internal, although it is not meant to be called by developers... although conceivably, it could be called manually if the developer has some reason to make the map-marker not a child of the map. My preference would be to not support that use case though. Can we mark this with an underscore even if it must be called from the Google maps component?

Reply:
Yeah, in our libraries we use underscore to indicate not only private, but that something is library-internal. This mainly comes about from needing to bind properties in templates that shouldn't be part of the public API.

Reply:
Done.

Reply:
Done.